specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2024-04-01-preview/fluxconfiguration.json

```diff
@@ -0,0 +1,1822 @@
+          "type": "string",
+          "x-nullable": true
+        }
+      }
+    },
+    "LayerSelectorDefinition": {
+      "description": "Parameters to specify which layer to pull from the OCI artifact. By default, the first layer in the artifact is pulled.",
+      "type": "object",
+      "x-nullable": true,
+      "properties": {
+        "mediaType": {
+          "description": "The first layer matching the specified media type will be used.",
+          "type": "string",
+          "x-nullable": true
+        },
+        "operation": {
+          "description": "The operation to be performed on the selected layer. The default value is 'extract', but it can be set to 'copy'.",
+          "type": "string",
+          "x-nullable": true
+        }
+      }
+    },
+    "LayerSelectorPatchDefinition": {
+      "description": "Parameters to specify which layer to pull from the OCI artifact. By default, the first layer in the artifact is pulled.",
+      "type": "object",
+      "x-nullable": true,
+      "properties": {
+        "mediaType": {
+          "description": "The first layer matching the specified media type will be used.",
+          "type": "string",
+          "x-nullable": true
+        },
+        "operation": {
+          "description": "The operation to be performed on the selected layer. The default value is 'extract', but it can be set to 'copy'.",
+          "type": "string",
+          "x-nullable": true
+        }
+      }
+    },
+    "VerifyDefinition": {
+      "description": "Parameters to verify the authenticity of an OCI Artifact.",
+      "type": "object",
+      "x-nullable": true,
+      "properties": {
+        "provider": {
+          "description": "Verification provider name.",
+          "type": "string"
+        },
+        "verificationConfig": {
+          "description": "An object containing trusted public keys of trusted authors.",
+          "type": "object",
+          "x-nullable": true,
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "matchOIDCIdentity": {
+          "description": "Array defining the criteria for matching the identity while verifying an OCI artifact.",
+          "type": "array",
+          "items": {
+            "type": "object",
+            "$ref": "#/definitions/MatchOidcIdentityDefinition"
+          },
+          "x-ms-identifiers": [],
+          "x-nullable": true
+        }
+      }
+    },
+    "VerifyPatchDefinition": {
+      "description": "Parameters to verify the authenticity of an OCI Artifact.",
+      "type": "object",
+      "x-nullable": true,
+      "properties": {
+        "provider": {
+          "description": "Verification provider name.",
+          "type": "string"
+        },
+        "verificationConfig": {
+          "description": "An object containing trusted public keys of trusted authors.",
+          "type": "object",
+          "x-nullable": true,
+          "additionalProperties": {
```

Comment:
>additionalProperties [](http://example.com/codeflow?start=11&length=20)

This is an ARM anti-pattern. Please schematize this object or model it as a collection.

Reply:
@mentat9, need additional properties as it is of the dictionary type and key and value are both not fixed. So cannot schematize this object. Can you please confirm what you mean by model it as a collection?

Reply:
Dictionaries are not allowed in Azure unless they satisfy the following criteria. For other situations, you may define it as a model object with concrete properties or an array of model.

1. The schema is not owned by your service
2. The keys of the dictionary are not validated by your service
3. The property represents an opaque property that is not interpreted by your service and is purely a pass thru.

Reply:
@ms-zhenhua, confirming that the keys of the dictionary are not validated by our service and are also simply passed through.